src/awx_plugins/credentials/aws_assumerole.py

```diff
@@ -0,0 +1,184 @@
+"""This module provides integration with AWS AssumeRole functionality."""
+
+import hashlib
+import typing
+from datetime import datetime
+
+from awx_plugins.interfaces._temporary_private_django_api import (  # noqa: WPS436
+    gettext_noop as _,
+)
+
+import boto3
+from botocore.exceptions import ClientError
+
+from .plugin import CredentialPlugin
+
+
+if typing.TYPE_CHECKING:
+    from mypy_boto3_sts.client import STSClient
+    from mypy_boto3_sts.type_defs import (
+        AssumeRoleResponseTypeDef,
+        CredentialsTypeDef,
+    )
+
+from .plugin import CredentialPlugin
+
+
+_aws_cred_cache: dict[
+    str,
+    'CredentialsTypeDef | dict[typing.Never, typing.Never]',
+] | dict[typing.Never, typing.Never] = {}
+
+
+assume_role_inputs = {
+    'fields': [
+        {
+            'id': 'access_key',
+            'label': _('AWS Access Key'),
+            'type': 'string',
+            'secret': True,
+            'help_text': _(
+                'The optional AWS access key for'
+                'the user who will assume the role.',
+            ),
+        },
+        {
+            'id': 'secret_key',
+            'label': 'AWS Secret Key',
+            'type': 'string',
+            'secret': True,
+            'help_text': _(
+                'The optional AWS secret key for the'
+                'user who will assume the role.',
+            ),
+        },
+        {
+            'id': 'external_id',
+            'label': 'External ID',
+            'type': 'string',
+            'help_text': _(
+                'The optional External ID which will'
+                'be provided to the assume role API.',
+            ),
+        },
+    ],
+    'metadata': [
+        {
+            'id': 'identifier',
+            'label': 'Identifier',
+            'type': 'string',
+            'help_text': _(
+                'The name of the key in the assumed AWS role'
+                'to fetch [AccessKeyId | SecretAccessKey | SessionToken].',
+            ),
+        },
+    ],
+    'required': [
+        'role_arn',
+    ],
+}
+
+
+def aws_assumerole_getcreds(
+        access_key: str,
+        secret_key: str,
```

Comment:
These two can still be `None` because that's what the boto3 function expects:
```suggestion
        access_key: str | None,
        secret_key: str | None,
```